.github/workflows/ci.yaml

```
@@ -15,19 +15,117 @@ concurrency:
   group: ${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}
   cancel-in-progress: true
 
-env:
-  DOCKER_IMAGE: hirosystems/${{ github.event.repository.name }}
-
 jobs:
+  api-lint:
+    strategy:
+      fail-fast: false
+      matrix:
+        suite: [ordinals]
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./api/${{ matrix.suite }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: 'api/${{ matrix.suite }}/.nvmrc'
+
+      - name: Cache node modules
+        uses: actions/cache@v4
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: |
+            ~/.npm
+            **/node_modules
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: Install deps
+        run: npm ci --audit=false
+
+      - name: Lint ESLint
+        run: npm run lint:eslint
+
+      - name: Lint Prettier
+        run: npm run lint:prettier
+
+      - name: Lint Unused Exports
+        run: npm run lint:unused-exports
+
+  api-test:
```

Should we run the test suite only if the linter passes? This would reduce the load on CI.

```suggestion
  api-test:
    needs: api-lint
```

---

I think leaving it separate is ok. We use this standard across all APIs so we can iterate a bit faster.